### ISSUE TITLE
Incorrect dataContext is passed in `onSave` callback using composite editor

### ISSUE THREAD

**Issue by rossmills99** [ghiscoding/Angular-Slickgrid#1333]
### Describe the bug

The `dataContext` property (third argument of the `onSave` callback on `CompositeEditorOpenDetailOption`) has the values of an existing row from the grid, instead of containing the new values that were entered using the composite editor.

### Reproduction

The `dataContext` appears to contain correct values after the first time of using it.

On the second and all subsequent attempts to use it, `dataContext` appears to contain the data of the row above/before the currently active cell of the grid.

### Expectation

`dataContext` argument should contain the data that were entered in the composite editor dialog.

### Environment Info

```shell
Angular ^16.2.0
angular-slickgrid ^6.6.4
@slickgrid-universal/composite-editor-component ^3.7.1
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Quick Start](https://ghiscoding.gitbook.io/angular-slickgrid/getting-started/quick-start).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
I cannot reproduce this on my side, the [Example 30](https://ghiscoding.github.io/Angular-Slickgrid/#/composite-editor) already has console log on the `onSave` and editing a single row (from Action menu "Edit Row" on the right), everything seems fine by me. Below is the live demo with editing 2nd row then 4th row and finally the 5th row, then you can see in the console that the data context is correct

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/643976/4e38fd4a-3084-411b-a255-8d4a723e621e)

Note that I have now published v7.0 a couple days ago and therefore I stopped supporting v6.x.

**Comment by rossmills99** [comment]
I've reproduced using the demo page, here I've just added "Task 102" but the data context has "Task 99":

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/5502815/17742c51-df17-4e75-87ea-690583edae54)

The issue occurs when creating new items in the grid, rather than editing existing items.  I may be misinterpreting the purpose for the `onSave` arguments mind you - I was using `dataContext` because I do not see any values passed in the first argument (`formValues`), but perhaps the dataContext means something different than what I was expecting?

**Comment by ghiscoding** [comment]
ahhh ok I can take a look at the insert, I actually never used it even though I added the feature. I had Cypress test for that feature but only for 1 insert which is probably why I didn't catch this issue.

**Comment by rossmills99** [comment]
Thanks for looking at that, and for the work you are open-sourcing with this project!

For my own use case, I think I will move away from using the composite editor and implement dialogs separately, just to have more control over the behaviour.  The great thing is there sees to be plenty of flexibility to do that.

**Comment by ghiscoding** [comment]
I pushed a fix in Slickgrid-Universal as referenced in the link above, this will be released in a couple days but will only be available for v7.x since I don't want to support older versions of Angular-Slickgrid/Slickgrid-Universal (it's too much work for a sole developer like me on these open source projects)

**Comment by ghiscoding** [comment]
I have reimplemented the `extend` function that was previously used and shows up in your stack trace, the new implementation follows [`node-extend`](https://www.npmjs.com/package/extend) package which is a more proven package, hopefully this fixes your issue. So give this another try and please tell it works 🤞🏻

**Comment by ghiscoding** [comment]
closed by #1337
